projects/npm-tools/packages/changelog-generator/src/index.js

```diff
@@ -749,6 +937,41 @@ async function main(_node, _script, ...args) {
 		}
 	}
 
+	try {
+		await go(options);
+	}
+	finally {
+		if (readline) {
+			readline.close();
+		}
+	}
+}
+
+/**
+ * Depending on the `options` in effect:
+ *
+ * - Prints `preview` to standard output (eg. during `--dry-run`); or:
+ * - Prints `contents` to the specified `outfile`.
+ */
+async function write(options, preview, contents) {
+	if (options.dryRun || options.version.endsWith(PLACEHOLDER_VERSION)) {
+		if (options.dryRun) {
+			info(`[--dry-run] Preview of changes 👀`);
+		}
+		else if (options.interactive) {
```

Comment:
These conditionals kind of suck, specifically how `options.interactive` explicitly represents a command-line option but the `endsWith()` check encodes application state (because in interactive mode we'll _usually_ start with a placeholder — although it's not compulsory — and then after we've seen the preview we'll have a non-placeholder version when we run again). Another thing to think of an improvement for... 🤔